enumerables.rb

```diff
@@ -1,88 +1,100 @@
+# frozen_string_literal: true
+
+# enumerable methods
 module Enumerable
   def my_each
     arr = self
+    if block_given?
       for i in arr
         yield(i)
+        puts i
       end
+    else
+      arr.to_enum(:my_each)
+    end
   end
 
   def my_each_with_index
     arr = self
-      for i in arr
+    if block_give?
+      arr.my_each do |i|
         v = arr.find_index(i)
         yield(i, v)
       end
+    else
+      arr.to_enum(:my_each_with_index)
+    end
   end
 
   def my_select
     arr = self
     result = []
-    for i in arr
-      if  yield i
-        result.push(i)
-        end
+    arr.each do |i|
+      result.push(i) if yield i
     end
     result
   end
 
-  def my_all(argument = nil)
+  def my_all?(arg = nil)
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false unless yield(i)
+        return false unless yield(i)
       end
     else
-      my_each { |i| return false if i.nil? || i == false }
+      my_each { |i| return false if !i.is_a? arg }
     end
     true
   end
-
-  def my_all(argument = nil)
+  def my_any?( arg = nil )
     arr = self
     if block_given?
-      arr.my_each do |i|
-      return false
+      arr.my_each do |i| 
+        return true if yield(i)
       end
     else
-      my_each { |i| yield i if i.nil? || i == false }
+      my_each { |i| return true if i.is_a? arg }
     end
-    true
+    false
   end
 
-  def my_none(argument = nil)
+  def my_none?( arg = nil )
     arr = self
     if block_given?
       arr.my_each do |i|
-      return false if yield i
+        return false if yield i
       end
     else
-      my_each { |i| yield i if i.nil? && i == false }
+      my_each { |i| return false if i.is_a? arg }
     end
     true
   end
 
   def my_map
     arr = self
-    mapped_arr =[]
-      arr.my_each do |i| i
-        mapped_arr.push(yield i)
-        yield i
-      end
-      mapped_arr
+    mapped_arr = []
+    arr.my_each do |i|
+      mapped_arr.push(yield i)
+      yield i
+    end
+    mapped_arr
   end
```

Comment:
* [ ] should have working `my_map` identical and returns the same thing as ruby's [`map`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-map). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_map.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

* [ ] my_map returns an Enumerator if no block is given
   ```ruby
    array = [1,5,8,7,9,5,5,55,5]
    p array.my_map #error
    LocalJumpError:
        no block given (yield)
  # ./enumerables.rb:77:in `block in my_map'
      # ./enumerables.rb:9:in `block in my_each'
      # ./enumerables.rb:8:in `each'
      # ./enumerables.rb:8:in `my_each'
      # ./enumerables.rb:76:in `my_map'
  ```